### ISSUE TITLE
[WISH] Method to cancel uploads

### ISSUE THREAD

**Issue by GoogleCodeExporter** [bdshadow/gwtupload#5]
```
Hi,

it would be really helpfull to allow the user to cancel uploads. Maybe you
could add some methods for this ?
```

Original issue reported on code.google.com by `norman.m...@googlemail.com` on 11 Aug 2009 at 5:49

**Comment by GoogleCodeExporter** [comment]
```
What do you mean with cancel upload, just a button to discard files waiting in 
the
queue, or to cancel the file being uploaded, or to discard all uploaded files 
in the
server?.

Thanks
Manolo
```

Original comment by `manuel.carrasco.m` on 18 Aug 2009 at 9:16
- Changed state: **Started**
- Added labels: **Feedback**
- Removed labels: **Type-Defect**

**Comment by GoogleCodeExporter** [comment]
```
Hi,

I would like to be able to cancel the current upload via some method (no 
button). But
having a button to remove uploaded files again whould be nice too.

Thx,
Norman
```

Original comment by `norman.m...@googlemail.com` on 18 Aug 2009 at 9:26

**Comment by GoogleCodeExporter** [comment]
```
Cancel files in the queue or delete uploaded files from the server is easy.

But cancel the current upload process is difficult. I don't know a way to 
cancel the
submit action from the client part. A possibility is to send an Ajax request to 
the
server to raise an exception, but this doesn't close the opened socket and the 
upload
continues until all the file content has uploaded.
Do you have any suggestion about how to do that?

```

Original comment by `manuel.carrasco.m` on 18 Aug 2009 at 10:42

**Comment by GoogleCodeExporter** [comment]
```
Unfortunally no.. I just would like to have the feature ;) Because I'm writting 
a
webmail client and it would be nice to cancel the current upload if the user 
dedicide
to drop the drafted email.
```

Original comment by `norman.m...@googlemail.com` on 18 Aug 2009 at 10:55

**Comment by GoogleCodeExporter** [comment]
```
Yes I know your project, I've sent you an invitation to use gtalk, so if you 
want we
could continue this chat in a faster way.
```

Original comment by `manuel.carrasco.m` on 18 Aug 2009 at 11:09

**Comment by GoogleCodeExporter** [comment]
```
Norman, 
I've a prototype working with this feature, please take a look to this examples 
and
let me know if this solution satisfies your requirements. 

http://gwtupload.alcala.org/gupld/gwtuploadsample.IncubatorUploadSample/Incubato
rUploadSample.html

http://gwtupload.alcala.org/gupld/gwtuploadsample.ChismesUploadSample/ChismesUpl
oadSample.html

Thanks
Manolo
```

Original comment by `manuel.carrasco.m` on 1 Sep 2009 at 10:31
- Added labels: **Feedbak**, **Type-Enhancement**
- Removed labels: **Feedback**

**Comment by GoogleCodeExporter** [comment]
```
Seems to be exactly what I need. Thx
```

Original comment by `norman.m...@googlemail.com` on 1 Sep 2009 at 5:11

**Comment by GoogleCodeExporter** [comment]
```
Hello Norman, I've released the version 0.5.1 where this feature has been added.

As I told you, I will send a patch to hupa in order to use the new gwtupload 
version.

Thanks
Manolo.
```

Original comment by `manuel.carrasco.m` on 13 Sep 2009 at 10:13
- Changed state: **Fixed**
- Removed labels: **Feedbak**

**Comment by GoogleCodeExporter** [comment]
```
Hi manuel,

thx for taking care of this. I'm looking forward to include your code to hupa :)
```

Original comment by `norman.m...@googlemail.com` on 13 Sep 2009 at 3:53
